### DISCUSSION TITLE
typing of unparsed SearchParams necessary?

### DISCUSSION THREAD

**Discussion by rwieruch** [47ng/nuqs#522] [General] [answered]
Sorry for keeping it short, my browser just lost my text from before :(

I have the following setup:

```
export type SearchParams = {
  search: string | string[] | undefined;
  sort: string | string[] | undefined;
};

export const searchParamsCache = createSearchParamsCache({
  search: parseAsString.withDefault(''),
  sort: parseAsString.withDefault(''),
});
```

And use the `searchParamsCache` the following way to operate on my Next page's `searchParams: SearchParams`:

```
searchParamsCache.parse(searchParams)
```

But I could give `SearchParams` any type,  `searchParamsCache.parse` would never complain.

So my question: Does it make sense to type `SearchParams` properly as I do or can I just set it to:

```
export type SearchParams = unknown;
```

What's the best practice here? Thanks! :)

**Comment by franky47** [comment, marked as answer]
The Next.js docs recommend specifying the keys you're interested in, but since you can technically receive anything, a more correct type would be `Record<string, string | string[] | undefined>`, which is conveniently exported by `nuqs`:

```tsx
// page.tsx
import type { SearchParams } from 'nuqs/server'

type PageProps = {
  searchParams: SearchParams
}

export default function Page({ searchParams }: PageProps) {
  // ...
}
```

**Reply by rwieruch** [reply]
Perfect! Thanks @franky47 <3
